ui/src/App.js

```diff
@@ -1,27 +1,57 @@
 import React, { Component } from 'react';
 import { Router, Route, Switch } from 'react-router-dom';
-import { FrontSite, BackOffice } from '@routes/components';
-import { BackOfficeRoutes } from '@routes/urls';
+import { connect } from 'react-redux';
+import { fetchUserProfile } from '@authentication/state/actions';
+import { FrontSite, BackOffice, Accounts } from '@routes/components';
+import { AccountsRoutes, BackOfficeRoutes } from '@routes/urls';
 import history from './history';
 import { NotFound } from '@components';
-import { AuthenticationGuard, UnAuthorized } from './authentication/components';
+import { AuthenticationGuard, UnAuthorized } from '@authentication/components';
 import PropTypes from 'prop-types';
 
+class SetUserInfoComponent extends Component {
+  componentDidMount() {
+    if (this.userWasPreviouslyLoggedIn) {
+      this.props.fetchUserProfile();
+    }
+  }
+
+  get userWasPreviouslyLoggedIn() {
+    return localStorage.getItem('ILS_USER_WAS_LOGGEDIN') || false;
```

Comment:
@equadon No this variable indicated if user was loggedin before accessing this page. For example, I visit the site I log in and then I close the browser. Next day that I visit the site again this variable it would have value true. But if I had logged out before I closed the browser and then I am re visiting the site then it would have value false and thus there is no need to fetch for user information.